### PR TITLE
(BOLT-608) Set provided features for local transport  

### PR DESCRIPTION
**Pull request by donoghuc**
When using the local transport set `provided_features` to `shell` or `powershell` to reflect running on a windows or POSIX system. 

Run the local transport unit tests on windows bolt controllers.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -36,6 +36,7 @@
   end
 
   config.filter_run_excluding appveyor_agents: true unless ENV['APPVEYOR_AGENTS']
+  config.filter_run_excluding windows: true unless ENV['BOLT_WINDOWS']
```
Comment (MikaelSmith) [inline spec/spec_helper.rb]: We generally do this via tags in the Rakefile, not separate environment variables.

Comment (MikaelSmith) [reply spec/spec_helper.rb]: Actually not sure why this is necessary at all.

File: spec/bolt/transport/shared_examples.rb [hunk 2]
```diff
@@ -36,6 +37,25 @@ def posix_context
   }
 end
 
+def windows_context
```
Comment (MikaelSmith) [inline spec/bolt/transport/shared_examples.rb]: Oh oops, I had a local copy of much of this stashed.

How does the stderr_command work?

Comment (donoghuc) [reply spec/bolt/transport/shared_examples.rb]: I think ssh client sends that info to `stderr` with default https://en.wikibooks.org/wiki/OpenSSH/Print_version config. I guess Appveyor must have `openssh`, the win10 floaty box I developed tests on does. 
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ irb
irb(main):001:0> require 'open3'
=> true
irb(main):002:0> Open3.capture3('ssh -V')
=> ["", "OpenSSH_7.2p2 Ubuntu-4ubuntu2.6, OpenSSL 1.0.2g  1 Mar 2016\n", #<Process::Status: pid 26948 exit 0>]
```

Comment (MikaelSmith) [reply spec/bolt/transport/shared_examples.rb]: I'd expect this test to fail with the Nano Server VM.

File: spec/spec_helper.rb (hunk 1 above)
Comment (MikaelSmith) [reply spec/spec_helper.rb]: Ah. I guess all the other tests run from macOS or Linux. Definitely a non-Windows view of development.

**Comment by MikaelSmith** [comment]
I'm comfortable merging this as-is unless you want to try to come up with an alternative to OpenSSH.

**Pull request merged**